public/app/features/logs/components/panel/LogLineContext.tsx

```diff
@@ -249,6 +314,38 @@ export const LogLineContext = memo(
         >
           <LogLineDetailsLog log={logListModel} syntaxHighlighting={syntaxHighlighting} />
         </Collapse>
+        <div className={styles.controls}>
+          {log.datasourceType === 'loki' && (
```

Comment:
Does this need to be loki only?

Reply:
Not really, as all these are DataSourceWithLogContextSupport.

We did the same for label types. https://github.com/grafana/grafana/blob/main/public/app/features/logs/utils.ts#L406

Reply:
I can provide a PoC for elastic to implement this, which will help document what needs to be done for other datasources as well. Seems pretty straight-forward to implement on the ds

Reply:
> I can provide a PoC for elastic to implement this, which will help document what needs to be done for other datasources as well. Seems pretty straight-forward to implement on the ds

Please do!